os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -712,7 +712,9 @@ int binary_manager_execute_loader(int cmd, int bin_idx)
 	int ret;
 	uint8_t loader_priority;
 	main_t loader_func;
-	char data_str[1];
+	/* int limits -2,147,483,648 to 2,147,483,647, 11 characters + 1 null character.
+	 * We need to allocate 12bytes atleast to prevent buffer overflows during itoa */
+	char data_str[12];
```

Comment:
`data_str` variable is used only for `bin_idx` which is a index of binary table and I think it will be maximum under 5. Is it needed to expand?